src/main/java/org/carlmontrobotics/commands/AlignToNoteMath.java

```diff
@@ -54,57 +54,140 @@ public class AlignToNoteMath extends ProxyCommand {
+                }
+
+                for (int i = 0; i < blueNoteX.length; i++) {
+                        blueNoteDistances[i] = Math.hypot(
+                                        dt.getPose().getX() - blueNoteX[i],
+                                        dt.getPose().getY() - blueNoteY[i]);
+                }
+
+                for (int i = 0; i < centerNoteX.length; i++) {
+                        centerNoteDistances[i] = Math.hypot(
+                                        dt.getPose().getX() - centerNoteX[i],
+                                        dt.getPose().getY() - centerNoteY[i]);
+                }
+
+                /*
+                 * double redNote1distance = Math.hypot(dt.getPose().getX() - rednote1X,
+                 * dt.getPose().getY() - rednote1Y); double redNote2distance =
+                 * Math.hypot(dt.getPose().getX() - rednote2X, dt.getPose().getY() - rednote2Y);
+                 * double redNote3distance = Math.hypot(dt.getPose().getX() - rednote3X,
+                 * dt.getPose().getY() - rednote3Y);
+                 * 
+                 * double blueNote1distance = Math.hypot(dt.getPose().getX() - bluenote1X,
+                 * dt.getPose().getY() - bluenote1Y); double blueNote2distance =
+                 * Math.hypot(dt.getPose().getX() - bluenote2X, dt.getPose().getY() - bluenote2Y);
+                 * double blueNote3distance = Math.hypot(dt.getPose().getX() - bluenote3X,
+                 * dt.getPose().getY() - bluenote3Y);
+                 * 
+                 * double Note4distance = Math.hypot(dt.getPose().getX() - note4X,
+                 * dt.getPose().getY() - note4Y); double Note5distance =
+                 * Math.hypot(dt.getPose().getX() - note5X, dt.getPose().getY() - note5Y); double
+                 * Note6distance = Math.hypot(dt.getPose().getX() - note6X, dt.getPose().getY() -
+                 * note6Y); double Note7distance = Math.hypot(dt.getPose().getX() - note7X,
+                 * dt.getPose().getY() - note7Y); double Note8distance =
+                 * Math.hypot(dt.getPose().getX() - note8X, dt.getPose().getY() - note8Y);
+                 */
+
+                // DISTANCES ARRAY
+                /*
+                 * double [] redNoteDistances = {redNote1distance, redNote2distance,
+                 * redNote3distance};
+                 * 
+                 * double [] blueNoteDistances = {blueNote1distance, blueNote2distance,
+                 * blueNote3distance};
+                 * 
+                 * double [] centerNoteDistances = {Note4distance, Note5distance, Note6distance,
+                 * Note7distance, Note8distance};
+                 */
+                for (int i = 0; i < redNoteDistances.length; i++) {
+                        if (redNoteDistances[i] < 2.0) {
+                                return new RotateToFieldRelativeAngle(
+                                                new Rotation2d(redNoteX[i] + dt
```

Comment:
Why is this a "+" for red notes but a "-" for blue notes?

Reply:
Because the robot intake is facing a different direction depending on an alliance, so we have to add or subtract robot pos from note pos

Reply:
Have you tested this?